js/Model.js

```diff
@@ -508,6 +508,7 @@ define(
 					this.currentCohortDefinitionInfo([]);
 				} else {
 					const { data: cohortDefinition } = await httpService.doGet(config.api.url + 'cohortdefinition/' + cohortDefinitionId);
+					cohortDefinition.expression = JSON.parse(cohortDefinition.expressionStr);
```

Comment:
1. Would be reasonable to encapsulate it in a service method
2. Now you have two methods for getting the `expression` and no description (comments) is provided what is the difference between those

Reply:
I removed this line because now cohort expression comes to UI already as Object